zeno/agents/maingraph/models.py

```diff
@@ -0,0 +1,100 @@
+import os
+from langchain_ollama import ChatOllama
+from langchain_anthropic import ChatAnthropic
+from langchain_openai import ChatOpenAI
+
+# TODO: combine the defintions for the OpenAI models
+# TODO: ChatAnthropic doesn't have a generic "json" type output. We can instead create an output parser class,
+
+MODELS_CONFIG = [
+    {
+        "required_env_var": "OLLAMA_BASE_URL",
+        "model_id": "llama3.2",
+        "model_name": "Ollama llama3.2",
+        "constructor_class": ChatOllama,
+        "json_mode": {"param_name": "format", "param_value": "json"},
+        "additional_params": {
+            "base_url": os.environ.get("OLLAMA_BASE_URL"),
+        },
+    },
+    {
+        "required_env_var": "OLLAMA_BASE_URL",
+        "model_id": "qwen2.5:7b",
+        "model_name": "Ollama Qwen2.5 (7 billion)",
+        "constructor_class": ChatOllama,
+        "json_mode": {"param_name": "format", "param_value": "json"},
+        "additional_params": {
+            "base_url": os.environ.get("OLLAMA_BASE_URL"),
+        },
+    },
+    # {
+    #     "required_env_var": "ANTHROPIC_API_KEY",
```

Comment:
Why is this commented out, it would be great to have Claude as an option.

Reply:
Anthropic doesn't have a `format="json"` parameter readily available on the `ChatAnthropic()` model instantiation method. We can force JSON output by using `llm.with_structured_output()` and pass an output definition, but since the various nodes/tools are already using `llm.with_structured_output()` _after_ having fetched the instantiated model from the `ModelFactory`, I was unsure of the potential effects of the double `with_structured_output()` call, so I disabled the Anthropic model class for now